File: 101/scala/src/test/scala/BinarySearchTest.scala
import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers

class BinarySearchTest extends AnyFunSuite with Matchers {
  private val TestArray = Seq(5, 10, 15, 20, 25)

  test("Binary search finds values") {
    BinarySearch(TestArray, 10) should be(true)
  }

  test("Binary search does not find value when not present") {
    BinarySearch(TestArray, 11) should be(false)
  }
}

File: 101/ts/test/binarySearch.test.ts
import {binarySearch} from '../src/binarySearch';

describe('BinarySearch', () => {
  const array = [5, 10, 15, 20, 25, 30, 35];

  it('should find item in array', () => {
    expect(binarySearch(array, 10)).toBeTruthy();
    expect(binarySearch(array, 30)).toBeTruthy();
  });

  it('should not find item in array', () => {
    expect(binarySearch(array, 11)).toBeFalsy();
  });
});
